FileCache race condition, ignoring error while copying, expected left-over temp copy file to be older...

```
FileCache: using existing file /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00213-of-00848.arrow
FileCache: using existing file /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00735-of-00848.arrow
FileCache: Copy file /rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMap
HuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00394-of-00848.arrow to cache
FileCache: Delete file /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-08-aed-large/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00398-of-00848.arrow, size 694.3MB. Still need more space, file is 19.9 hours old. After deletion, have 1.2GB free space.
FileCache: Copy file /rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMap
HuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00822-of-00848.arrow to cache
FileCache: lock for /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-08-aed-large/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00473-of-00848.arrow is currently held, skipping.
FileCache: Delete file /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-08-aed-large/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00754-of-00848.arrow, size 288.6MB. Still need more space, file is 19.9 hours old. After deletion, have 571.4MB free space.
FileCache: using existing file /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00367-of-00848.arrow
FileCache: lock for /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-08-aed-large/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00503-of-00848.arrow is currently held, skipping.
FileCache: Copy file /rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow to cache
FileCache: Ignoring error while copying /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow: OSError: [Errno 28] No space left on device
```

In another process, at the same time, I got this:
```
Process DistributeFilesDataset train ep 70 worker ep 70:
EXCEPTION
Traceback (most recent call last):
  File "/usr/lib64/python3.12/multiprocessing/process.py", line 314, in BaseProcess._bootstrap
    line: self.run()
    locals:
      self = <local> <NonDaemonicSpawnProcess name='DistributeFilesDataset train ep 70 worker ep 70' parent=488525 started>
  File "/usr/lib64/python3.12/multiprocessing/process.py", line 108, in BaseProcess.run
    line: self._target(*self._args, **self._kwargs)
    locals:
      self = <local> <NonDaemonicSpawnProcess name='DistributeFilesDataset train ep 70 worker ep 70' parent=488525 started>
      self._target = <local> <function returnn.datasets.distrib_files._worker_proc_loop>
      self._args = <local> (70, 1, {'class': 'MultiProcDataset', 'dataset': {'class': 'HuggingFaceDataset', 'dataset_opts': [CachedFile(filename='/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00502...
      self._kwargs = <local> {}
  File "/rwthfs/rz/cluster/home/az668407/setups/combined/2021-05-31/tools/returnn/returnn/datasets/distrib_files.py", line 681, in _worker_proc_loop
    line: dataset = init_dataset(dataset_dict)
...
  File "/rwthfs/rz/cluster/home/az668407/setups/combined/2021-05-31/tools/returnn/returnn/util/file_cache.py", line 343, in FileCache
.handle_cached_files_in_config.<locals>._handle_value
    line: res = self.get_file(value.filename)
    locals:
      self = <local> <returnn.util.file_cache.FileCache object at 0x1490ac3154f0>
      value = <local> CachedFile(filename='/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow')
      value.filename = <local> '/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow', len = 193
  File "/rwthfs/rz/cluster/home/az668407/setups/combined/2021-05-31/tools/returnn/returnn/util/file_cache.py", line 121, in FileCache.get_file
    line: self._copy_file_if_needed(src_filename, dst_filename)
    locals:
FileCache: Copy file /rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow to cache
      self = <local> <returnn.util.file_cache.FileCache object at 0x1490ac3154f0>
      src_filename = <local> '/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow', len = 193
      dst_filename = <local> '/var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow', len = 230
  File "/rwthfs/rz/cluster/home/az668407/setups/combined/2021-05-31/tools/returnn/returnn/util/file_cache.py", line 418, in FileCache._copy_file_if_needed
    line: assert dst_tmp_file_age > self._lock_timeout * 0.8, (
              f"FileCache: Expected left-over temp copy file {dst_tmp_filename}"
              f" from crashed previous copy attempt"
              f" to be older than {self._lock_timeout * 0.8:.1f}s but it is {dst_tmp_file_age} seconds old"
          )
    locals:
      dst_tmp_file_age = <local> 1.007828950881958
      self = <local> <returnn.util.file_cache.FileCache object at 0x1490ac3154f0>
      self._lock_timeout = <local> 20
      dst_tmp_filename = <local> '/var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow.c
opy', len = 235

AssertionError: FileCache: Expected left-over temp copy file /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow.copy from crashed previous copy attempt to be older than 16.0s but it is 1.007828950881958 seconds old
```

Note, looking for `data-00012-of-00848.arrow` specifically in the log, there is:
`FileCache: Copy file /rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow to cache`
and then `FileCache: Ignoring error while copying /var/tmp//az668407/returnn/file_cache/rwthfs/rz/cluster/home/az668407/setups/2025-10-21-chunked-ctc/work/i6_core/datasets/huggingface/TransformAndMapHuggingFaceDatasetJob.FxPUVJtw1EeN/output/dataset/train/data-00012-of-00848.arrow: OSError: [Errno 28] No space left on device`.

(cc @NeoLegends)

--- a/returnn/util/file_cache.py
+++ b/returnn/util/file_cache.py
@@ -436,7 +436,10 @@ class FileCache:
                         os.remove(dst_tmp_filename)
                     except FileNotFoundError:
                         pass
-                    os.remove(info_file_name)
+                    try:
+                        os.remove(info_file_name)
+                    except FileNotFoundError:  # not really expected here, but safe to ignore
+                        pass
                     raise
                 os.rename(dst_tmp_filename, dst_filename)
 

